src/SqlAsyncCollector.cs

```diff
@@ -189,6 +190,7 @@ private async Task UpsertRowsAsync(IEnumerable<T> rows, SqlAttribute attribute,
                 }
 
                 TelemetryInstance.TrackEvent(TelemetryEventName.UpsertStart, props);
+                this._logger.LogDebugWithThreadId("BEGIN upsert rows transaction");
```

Comment:
nit: I'd suggest using a single word after BEGIN/END - the idea there was to make it easy to match the two events by searching for that (which is easier to do with just a single word you can double click to select)